src/enrichers/decisions.ts

```diff
@@ -0,0 +1,74 @@
+import { Enricher, EventBus } from '../types'
+import { getQueryParameter, ParsedParam } from '../utils/url'
+import { trim, isUUID, expiresInDays } from 'live-connect-common'
+import { WrappedStorageHandler } from '../handlers/storage-handler'
+
+type Input = { pageUrl?: string, domain: string }
+type Output = { decisionIds: string[] }
+
+const DEFAULT_DECISION_ID_COOKIE_EXPIRES = expiresInDays(30)
+const DECISION_ID_QUERY_PARAM_NAME = 'li_did'
+const DECISION_ID_COOKIE_NAMESPACE = 'lidids.'
+
+const _onlyUnique = (value: string, index: number, self: string[]) => self.indexOf(value) === index
+const _nonEmpty = (value: string) => value && trim(value).length > 0
+
+export function enrichDecisionIds(
+  storageHandler: WrappedStorageHandler,
+  eventBus: EventBus
+): Enricher<Input, Output> {
+  return state => {
+    function _addDecisionId(key: string, cookieDomain?: string) {
+      if (key) {
+        storageHandler.setCookie(
```

Comment:
I didn't want to touch it here, but in principle this should also go into the cache imo